projects/kit/components/multi-select-option/multi-select-option.component.ts

```diff
@@ -18,7 +18,13 @@ export class TuiMultiSelectOptionComponent<T> extends TuiSelectOptionComponent<T
         return (
             tuiIsPresent(this.option.value) &&
             tuiIsPresent(this.control.value) &&
-            !!this.control.value.find((item: T) => this.matcher(item, this.option.value!))
+            !!this.control.value.find((item: T) =>
```

Comment:
```ts
    protected override get selected(): boolean {
        const {value} = this.option;

        return (
            tuiIsPresent(value) &&
            tuiIsPresent(this.control.value) &&
            this.control.value.some((item: T) => this.matcher(item, value))
        );
    }
```
Note the change from `find` to `some`, otherwise falsy values give false negative results.

Reply:
fixed